workitem/expression_compiler.go

```diff
@@ -394,6 +406,51 @@ func (c *expressionCompiler) Not(e *criteria.NotExpression) interface{} {
 	return c.binary(e, "!=")
 }
 
+func (c *expressionCompiler) Child(e *criteria.ChildExpression) interface{} {
+	left, ok := e.Left().(*criteria.FieldExpression)
+	if !ok {
+		c.err = append(c.err, errs.Errorf("invalid left expression (not a field expression): %+v", e.Left()))
+		return nil
+	}
+
+	if strings.Contains(left.FieldName, "'") {
+		// beware of injection, it's a reasonable restriction for field names,
+		// make sure it's not allowed when creating wi types
+		c.err = append(c.err, errs.Errorf("single quote not allowed in field name: %s", left.FieldName))
+		return nil
+	}
+	litExp, ok := e.Right().(*criteria.LiteralExpression)
+	if !ok {
+		c.err = append(c.err, errs.Errorf("failed to convert right expression to literal expression: %+v", e.Right()))
+		return nil
+	}
+	r, ok := litExp.Value.(string)
+	if !ok {
+		c.err = append(c.err, errs.Errorf("failed to convert value of right literal expression to string: %+v", litExp.Value))
+		return nil
+	}
+
+	var tblName string
+	if left.FieldName == SystemIteration {
```

Comment:
Please don't introduce more places of non-generic code.

Reply:
This "child" (or "SubPathMatch") is only going to work for iteration and area. I am not sure how to generalize it.